src/components/Countries.js

```diff
@@ -0,0 +1,72 @@
+import { useSelector, useDispatch } from 'react-redux';
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { getCountries } from '../redux/countries/countrySlice';
+import Search from './Search';
+import africa from '../africa-map.png';
+
+function Countries() {
+  const dispatch = useDispatch();
+  const countriesArr = useSelector((state) => state.countries.countrydata);
+  const status = useSelector((state) => state.countries.status);
+  const [filteredCountries, setFilteredCountries] = useState([]);
+  const [searchValue, setSearchValue] = useState('');
+
+  useEffect(() => {
+    dispatch(getCountries());
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (countriesArr) {
+      setFilteredCountries(
+        countriesArr.filter((country) => country.name
+          .toLowerCase().includes(searchValue.toLowerCase())),
+      );
+    }
+  }, [countriesArr, searchValue]);
+
+  const handleSearch = (value) => {
+    setSearchValue(value);
+  };
+
+  if (status === 'idle' || status === 'pending' || !countriesArr) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div className="Main-div">
+      <div className="Top-section">
+        <img className="Africa-map" src={africa} alt="map of africa" />
+        <div className="Second-part">
+          <h2>Africa</h2>
+          <Search onSearch={handleSearch} />
+        </div>
+      </div>
+      <div className="Countries-grid">
+        {searchValue && filteredCountries.length === 0 ? (
+          <div>No countries found.</div>
+        ) : (
+          filteredCountries.map((country) => (
+            <div className="Country-list" key={country.id}>
+              <Link className="Country-link" to={`/country/${country.id}`} key={country.id}>
+                <div
+                  className="Flag-div"
+                  style={{
+                    backgroundImage: `url(${country.image})`,
+                    backgroundSize: 'cover',
+                    backgroundRepeat: 'no-repeat',
+                    position: 'relative',
+                  }}
+                >
+                  <h4 className="Country-name">{country.name}</h4>
```

Comment:
- hey, according to the main design, the color here is white and the country name should be aligned to the right and not at the center as you've done. kindly check the design to improve on this.
- Also, kindly add the missing arrow to improve the user experience of your page so that users can the contents in each card is clickable.

*main design*
![image](https://github.com/toyman640/metrics-webapp-capstone/assets/59397159/1c71bda3-a783-4fcc-9783-fed2481a1e02)